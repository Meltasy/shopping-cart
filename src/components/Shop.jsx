import ProductItems from "./ProductItems"

const Shop = () => {

  return (
      <div>
        <h1>Atelier</h1>
        <ProductItems />
      </div>
    )
}

export default Shop
