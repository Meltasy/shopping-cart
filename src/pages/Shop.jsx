import ShopItem from '../components/ShopItem'

const Shop = ({ productItems, handleSubmit }) => {

  return (
    <div>
      <h1>Atelier</h1>
      <div className='productItems'>
        {productItems.map(item => (
          <ShopItem
            key={item.id}
            itemId={item.id}
            title={item.title}
            price={item.price}
            imageUrl={item.image}
            description={item.description}
            onSubmit={handleSubmit}
          />
        ))}
      </div>
    </div>
  )
}

export default Shop
