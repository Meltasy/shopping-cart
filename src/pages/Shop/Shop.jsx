import ShopItem from '../../components/ShopItem/ShopItem'

const Shop = ({ productItems, handleAdd }) => {

  return (
    <div>
      <h1>Products</h1>
      <div className='productItems'>
        {productItems.map(item => (
          <ShopItem
            key={item.id}
            itemId={item.id}
            imageUrl={item.image}
            title={item.title}
            price={item.price}
            description={item.description}
            handleAdd={handleAdd}
          />
        ))}
      </div>
    </div>
  )
}

export default Shop
