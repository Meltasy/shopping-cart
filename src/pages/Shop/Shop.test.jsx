import { describe, it, expect, beforeEach } from 'vitest';
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import Shop from './Shop'

describe('Shop component', () => {
  const mockAdd = vi.fn()
  const mockProductItems = [
    {
      id: '123',
      image: 'fakeImageA.jpg',
      title: 'Fake product A',
      price: 35.50,
      description: 'This is fake product A.'
    }, {
      id: '321',
      image: 'fakeImageB.jpg',
      title: 'Fake product B',
      price: 57.99,
      description: 'This is fake product B.'
    }, {
      id: '231',
      image: 'fakeImageC.jpg',
      title: 'Fake product C',
      price: 7.50,
      description: 'This is fake product C.'
    },
  ]

  beforeEach(() => {
    render(<Shop productItems={mockProductItems} handleAdd={mockAdd} />)
  })

  it('renders shop products', () => {
    const { container } = render(<Shop productItems={mockProductItems} />)
    expect(container).toMatchSnapshot()
  })

  it ('renders correct heading', () => {
    screen.debug()
    expect(screen.getByRole('heading', { level: 1 }).textContent).toMatch(/products/i)
    expect(screen.getByRole('heading', { level: 1, name: 'Products' })).toBeInTheDocument()
  })

  it('renders shop products with all complete details', () => {
    expect(screen.getAllByRole('img')[0]).toHaveAttribute('src', 'fakeImageA.jpg')
    expect(screen.getAllByRole('heading', { hidden: true, level: 2 })[0]).toHaveTextContent('Fake product A â‚¬35.50')
    expect(screen.getByText('This is fake product A.')).toBeInTheDocument()
  })

  it('renders correct number of shop products', () => {
    const productItems = screen.getAllByRole('img')
    expect(productItems).toHaveLength(mockProductItems.length)
  })

  it('calls handleAdd on shop prodcut forms', async () => {
    const user = userEvent.setup()
    const addBtn = screen.getAllByRole('button', { name: 'Add to Cart' })[0]
    await user.click(addBtn)
    expect(mockAdd).not.toHaveBeenCalled()
  })
})
