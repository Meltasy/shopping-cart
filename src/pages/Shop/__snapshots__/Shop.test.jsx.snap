// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop component > renders shop products 1`] = `
<div>
  <h1>
    Products
  </h1>
  <div
    class="sc-cMqPwU keFeiM"
  >
    <div
      class="sc-kfeOyU cPZBqz"
    >
      <img
        alt=" "
        class="sc-hRDKVd fmIcTT"
        src="fakeImageA.jpg"
      />
      <h3>
        Fake product A
         €
        35.50
      </h3>
      <button
        class="sc-jBIHhB kKjMVe"
      >
        Show Detail
      </button>
      <dialog
        class="sc-FEMpB fROXOS"
      >
        <div
          class="sc-gjcoXW hEnKnr"
        >
          <h2>
            Fake product A
             €
            35.50
          </h2>
          <img
            alt=" "
            class="sc-dVBluf gNBXRn"
            src="fakeImageA.jpg"
          />
          <p>
            This is fake product A.
          </p>
          <button
            class="sc-jBIHhB kKjMVe"
          >
            Back to Shop
          </button>
        </div>
      </dialog>
      <form
        class="sc-fLDLck geSOOS"
      >
        <label
          class="sc-gNZgCX iOhCWI"
          for="quantity"
        >
          Quantity:
        </label>
        <div>
          <button
            class="sc-dycHyt hRrA-Dl"
            type="button"
          >
            -
          </button>
          <input
            class="sc-dycHyt sc-jbAkgO hRrA-Dl ilAZys"
            id="quantity"
            max="10"
            min="1"
            name="quantity"
            pattern="^\\d+$"
            type="text"
            value="0"
          />
          <button
            class="sc-dycHyt hRrA-Dl"
            type="button"
          >
            +
          </button>
        </div>
        <button
          class="sc-dycHyt hRrA-Dl"
          type="submit"
        >
          Add to Cart
        </button>
      </form>
    </div>
    <div
      class="sc-kfeOyU cPZBqz"
    >
      <img
        alt=" "
        class="sc-hRDKVd fmIcTT"
        src="fakeImageB.jpg"
      />
      <h3>
        Fake product B
         €
        57.99
      </h3>
      <button
        class="sc-jBIHhB kKjMVe"
      >
        Show Detail
      </button>
      <dialog
        class="sc-FEMpB fROXOS"
      >
        <div
          class="sc-gjcoXW hEnKnr"
        >
          <h2>
            Fake product B
             €
            57.99
          </h2>
          <img
            alt=" "
            class="sc-dVBluf gNBXRn"
            src="fakeImageB.jpg"
          />
          <p>
            This is fake product B.
          </p>
          <button
            class="sc-jBIHhB kKjMVe"
          >
            Back to Shop
          </button>
        </div>
      </dialog>
      <form
        class="sc-fLDLck geSOOS"
      >
        <label
          class="sc-gNZgCX iOhCWI"
          for="quantity"
        >
          Quantity:
        </label>
        <div>
          <button
            class="sc-dycHyt hRrA-Dl"
            type="button"
          >
            -
          </button>
          <input
            class="sc-dycHyt sc-jbAkgO hRrA-Dl ilAZys"
            id="quantity"
            max="10"
            min="1"
            name="quantity"
            pattern="^\\d+$"
            type="text"
            value="0"
          />
          <button
            class="sc-dycHyt hRrA-Dl"
            type="button"
          >
            +
          </button>
        </div>
        <button
          class="sc-dycHyt hRrA-Dl"
          type="submit"
        >
          Add to Cart
        </button>
      </form>
    </div>
    <div
      class="sc-kfeOyU cPZBqz"
    >
      <img
        alt=" "
        class="sc-hRDKVd fmIcTT"
        src="fakeImageC.jpg"
      />
      <h3>
        Fake product C
         €
        7.50
      </h3>
      <button
        class="sc-jBIHhB kKjMVe"
      >
        Show Detail
      </button>
      <dialog
        class="sc-FEMpB fROXOS"
      >
        <div
          class="sc-gjcoXW hEnKnr"
        >
          <h2>
            Fake product C
             €
            7.50
          </h2>
          <img
            alt=" "
            class="sc-dVBluf gNBXRn"
            src="fakeImageC.jpg"
          />
          <p>
            This is fake product C.
          </p>
          <button
            class="sc-jBIHhB kKjMVe"
          >
            Back to Shop
          </button>
        </div>
      </dialog>
      <form
        class="sc-fLDLck geSOOS"
      >
        <label
          class="sc-gNZgCX iOhCWI"
          for="quantity"
        >
          Quantity:
        </label>
        <div>
          <button
            class="sc-dycHyt hRrA-Dl"
            type="button"
          >
            -
          </button>
          <input
            class="sc-dycHyt sc-jbAkgO hRrA-Dl ilAZys"
            id="quantity"
            max="10"
            min="1"
            name="quantity"
            pattern="^\\d+$"
            type="text"
            value="0"
          />
          <button
            class="sc-dycHyt hRrA-Dl"
            type="button"
          >
            +
          </button>
        </div>
        <button
          class="sc-dycHyt hRrA-Dl"
          type="submit"
        >
          Add to Cart
        </button>
      </form>
    </div>
  </div>
</div>
`;
